src/behavior-considered-undefined.md

```diff
@@ -58,7 +58,10 @@ Please read the [Rustonomicon] before writing unsafe code.
 * Invoking undefined behavior via compiler intrinsics.
 * Executing code compiled with platform features that the current platform
   does not support (see [`target_feature`]), *except* if the platform explicitly documents this to be safe.
-* Calling a function with the wrong call ABI or unwinding from a function with the wrong unwind ABI.
+* Calling a function with the wrong [call ABI][abi].
```

Comment:
So there's no way at all to cause unwinding UB inside of Rust?

Reply:
Only with mismatched runtimes linked together (where it has the same problem as linking against any other "foreign" runtime); not within a single Rust binary.

Reply:
That's also, incidentally, why this was listed together with "calling a function with the wrong call ABI", as those things seem closely related when looking at it just from within Rust: caller and callee each have an idea of whether the call can unwind, and if they don't agree, that can cause UB. (But it's only UB if the unwind actually happens, the ABIs are otherwise compatible.)

Reply:
If I'm understanding the example correctly, fixing that issue is the point of the change to `extern "C"` behavior being stabilized: https://github.com/rust-lang/rust/issues/115285

We don't have a test case with a `transmute` being used to "hide" the real ABI, but I don't see how that would be different from using `extern "C"` to call a C++ function that unwinds, unless the compiler has an incorrect optimization here that assumes the `extern "C"` function pointer can't ever unwind for some reason.

I've written a quick test case here: https://github.com/rust-lang/rust/pull/129319

Reply:
I don't follow. The MIR terminator that calls `unwind()` has the `UnwindAction::Unreachable` set. That's unambiguously UB. Are you saying the MIR we are generating is wrong?

Reply:
I did think the MIR was wrong (i.e. I thought this was now well-defined), but you're correct, this is UB.

I've tried to make the bullet very explicit now about how, in pure Rust, we can still unwind incorrectly.